src/main/java/net/hypixel/nerdbot/command/GeneratorCommands.java

```diff
@@ -658,6 +665,186 @@ public void viewHistory(GuildSlashEvent event) {
+        @AppOption(description = "Determines if the `" + BASE_COMMAND + " item full` is included.") @Optional Boolean includeGenFullCommand,
+        @AppOption(description = "Changes the text at the bottom to display if its selected or not.") @Optional Boolean selected,
+        @AppOption(description = HIDDEN_OUTPUT_DESCRIPTION) @Optional Boolean hidden
+    ) {
+        if (hidden == null) {
+            hidden = getUserAutoHideSetting(event);
+        }
+        event.deferReply(hidden).complete();
+
+        alpha = alpha == null ? 245 : alpha;
+        padding = padding == null ? 0 : padding;
+
+        // Function to parse stats
+        Function<String, HashMap<String, Integer>> parseStatsToMap = stats -> {
+            HashMap<String, Integer> map = new HashMap<>();
+            String[] entries = stats.split(",");
+            for (String entry : entries) {
+                String[] stat = entry.split(":");
+                if (stat.length != 2 || stat[0].trim().isEmpty() || stat[1].trim().isEmpty()) {
+                    throw new GeneratorException("Invalid stats format: '" + entry + "'. Ensure each stat is in the format 'statName:integerValue'.");
+                }
+                String statName = stat[0].trim();
+                if (map.containsKey(statName)){
+                    throw new GeneratorException("Invalid stats list: Ensure you only have the same type of stat once. Duplicate stat: `" + statName + "`");
+                }
+
+                Integer statValue;
+                try {
+                    statValue = Integer.parseInt(stat[1].trim());
+                } catch (NumberFormatException e) {
+                    throw new GeneratorException("Invalid number format for stat '" + statName + "'. Ensure all values are integers.");
+                }
+                map.put(statName, statValue);
+            }
+            return map;
+        };
+
+        try {
+            DecimalFormat bigNumFormat = new DecimalFormat("#,###");
+
+            String scalingStatsFormatted = "";
+            for (Map.Entry<String, Integer> entry : (scalingStats != null ? parseStatsToMap.apply(scalingStats) : new HashMap<String, Integer>()).entrySet()){
+                String statName = entry.getKey();
+                Integer basePower = entry.getValue();
+                Stat stat = Stat.byName(statName);
+                if (stat == null){
+                    throw new GeneratorException("'" + statName + "' isn't a stat. Are you sure you typed it correctly?");
+                }
+
+                double statMultiplier = stat.getPowerScalingMultiplier() != null ? Stat.byName(statName).getPowerScalingMultiplier() : 1;
+
+                scalingStatsFormatted += String.format("%%%%%s:%s%%%%\\n",
+                    statName,
+                    bigNumFormat.format(Math.round(((double)basePower/100)*statMultiplier*719.28*Math.pow(Math.log(1+(0.0019*magicalPower)), 1.2)))
```

Comment:
Can we move whatever these numbers are to variables that explain what they are?

Reply:
these numbers arent really anything they are just constants used in the formula and don't have any real meaning.

Reply:
```suggestion
                    bigNumFormat.format(Math.round(((double)basePower/100)*statMultiplier*719.28*Math.pow(Math.log(1+(0.0019*magicalPower)), 1.2))) // Formula: https://imgur.com/a/QwckMcx Source: https://wiki.hypixel.net/Powers#Stat_Calculation
```

Reply:
I think the source link is fine on its own

Reply:
Can get outdated tho. Ig i could just directly link to the history (but then in the history the current file page) and then it should be fine?

Reply:
I honestly think it's fine because we'll just have to update the calculation in code ourselves